src/libical/icalcomponent.c

```diff
@@ -134,6 +134,9 @@ icalcomponent *icalcomponent_clone(const icalcomponent *old)
     icalerror_check_arg_rz((old != 0), "component");
 
     clone = icalcomponent_new_impl(old->kind);
+    if (old->x_name) {
```

Comment:
seems to me that this conditional needs to be moved below the `if (clone == 0)` immediately following

Reply:
Thanks, fixed.